master/internal/api_templates.go

```diff
@@ -42,16 +43,62 @@ func (a *apiServer) GetTemplate(
 	}
 }
 
-func (a *apiServer) PutTemplate(
-	_ context.Context, req *apiv1.PutTemplateRequest,
-) (*apiv1.PutTemplateResponse, error) {
+func (a *apiServer) PostTemplate(
+	ctx context.Context, req *apiv1.PostTemplateRequest,
+) (*apiv1.PostTemplateResponse, error) {
 	config, err := protojson.Marshal(req.Template.Config)
 	if err != nil {
 		return nil, status.Errorf(codes.InvalidArgument, "invalid config provided: %s", err.Error())
 	}
-	err = a.m.db.QueryProto("put_template", req.Template, req.Template.Name, config)
-	return &apiv1.PutTemplateResponse{Template: req.Template},
-		errors.Wrapf(err, "error putting template")
+	workspaceID := model.DefaultWorkspaceID
+	if req.Template.WorkspaceId != 0 {
+		workspaceID = int(req.Template.WorkspaceId)
+	}
+	notFoundErr := status.Errorf(codes.NotFound, "workspace (%d) not found", workspaceID)
+	var exists bool
+	err = db.Bun().NewSelect().ColumnExpr("1").Table("workspaces").
+		Where("id = ?", workspaceID).
+		Where("archived = false").
+		Limit(1).
+		Scan(ctx, &exists)
+	if err != nil {
+		return nil, errors.Wrapf(err, "error checking workspace %d", workspaceID)
+	}
+	if !exists {
+		return nil, notFoundErr
+	}
+
+	res := apiv1.PostTemplateResponse{Template: &templatev1.Template{}}
+	switch err := a.m.db.QueryProto(
+		"put_template", res.Template, req.Template.Name, config, workspaceID,
+	); err {
+	case nil:
+		return &res, nil
+	default:
+		return nil, status.Errorf(codes.Internal, "error posting template %s to db: %s",
+			req.Template.Name, err.Error())
+	}
+}
+
+func (a *apiServer) PatchTemplateConfig(
```

Comment:
should this be designed more as a generic patch endpoint which for now just allows setting the config? communicating that through protobuf could be tricky